src/Model/Entity/BaseEntity.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace Friendica\Model\Entity;
+
+use ArrayAccess;
+use Friendica\Network\HTTPException;
+
+abstract class BaseEntity implements ArrayAccess
```

Comment:
This is the base class of every entities of Friendica.
Means it is used for:
- `Business objects`: Introduction, Item, Contact, ...
- `Data Transfer Objects`: Mastodon Instance, Mastodon Account

It has the very basic functionality, which each entity in Friendica must have.

Reply:
Why don't you make it extend a `DataTransferObject` abstract class that would be what currently is `DataTransferObjectTrait`?

Reply:
Why not use `DataTransferObjectTrait` as an intermediate class between `BaseEntity` and `Instance`?

For example
```php
abstract class BaseEntity implements ArrayAccess;

abstract class DataTransferEntity extends BaseEntity;
abstract class StorableEntity extends BaseEntity;

class Instance extends DataTransferEntity;
class Contact extends StorableEntity;
```
I personally don't like using traits when you can use class inheritance.

Reply:
I wasn't sure, if they remain just these two traits. Maybe there is particular other functionallity for entities. But you're probably right